### PR TITLE
mock action performer configs

### PR DESCRIPTION
**Pull request by jeberl**
Summary
---------

Initial mock of the get_action_performer function. This will eventually read configs from s3 and dynamically determine which performer to use 

Test Plan
---------

1. go to https://webhook.site/ to get a unique URL
2.  add to the WebhookActionPerformer constructor
3. run `python3 -m hmalib.lambdas.actions.action_performer`
4. check that the webhook was recieved

### CODE REVIEW

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py [hunk 1]
```diff
@@ -22,8 +28,20 @@ def perform_enque_for_review(match_message: MatchMessage):
...
 
 def perform_action(match_message: MatchMessage, action_label: ActionLabel):
-    # TODO implement
-    logger.debug("perform action")
+    action_performer = get_action_perfromers_config()[action_label]
+    action_performer.perform_action(match_message)
+
+
+def get_action_perfromers_config() -> t.Dict[ActionLabel, ActionPerformer]:
+    # TODO Should Read From s3 Configs table and determine which performer dynamically
```
Comment (Dcallies) [inline hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:36]: I built the dynamodb config thing for these types of usecases, probably overall easier for same general results.

File: hasher-matcher-actioner/hmalib/common/actioner_models.py [hunk 2]
```diff
@@ -33,3 +38,51 @@ class ActionRule:
     action_label: ActionLabel
     must_have_labels: t.List[Label]
     must_not_have_labels: t.List[Label]
+
+
+TUrl = t.Union[t.Text, bytes]
```
Comment (Dcallies) [inline hasher-matcher-actioner/hmalib/common/actioner_models.py:43]: There might be a specific type for this buried in the typing module

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py [hunk 3]
```diff
@@ -22,8 +28,20 @@ def perform_enque_for_review(match_message: MatchMessage):
 
 
 def perform_action(match_message: MatchMessage, action_label: ActionLabel):
-    # TODO implement
-    logger.debug("perform action")
+    action_performer = get_action_perfromers_config()[action_label]
+    action_performer.perform_action(match_message)
+
+
+def get_action_perfromers_config() -> t.Dict[ActionLabel, ActionPerformer]:
```
Comment (Dcallies) [inline hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:35]: I like this approach, it's a great way to define the interfaces and unblock work, while someone else can iterate on the storage and config separately.

**Review by Dcallies — COMMENTED** [review]
I will leave approval and review to @eriksendc , I'm still struggling with the fact that I don't agree with the with the overall approach, and even though I had some suspicions originally it was because I didn't understand the proposal, after reading the code, I'm still hung on the same things.

 I know you are working with Brian on this, and I will defer to you to you two as the owners of the actioning to figure that out.

File: hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py (hunk 1 above)
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/lambdas/actions/action_performer.py:36]: Yeah. I figured I would implement once your PR merges

File: hasher-matcher-actioner/hmalib/common/actioner_models.py [hunk 4]
```diff
@@ -33,3 +38,51 @@ class ActionRule:
...
+
+class Post(HTTPRequestTypes):
+    @classmethod
+    def call(cls, url: TUrl, data: str) -> Response:
+        return post(url, data)
+
+
+class Get(HTTPRequestTypes):
+    @classmethod
+    def call(cls, url: TUrl, _data: str) -> Response:
```
Comment (eriksendc) [inline hasher-matcher-actioner/hmalib/common/actioner_models.py:60]: nix "unused argument" `_data`.

File: hasher-matcher-actioner/hmalib/common/actioner_models.py [hunk 5]
```diff
@@ -33,3 +38,51 @@ class ActionRule:
...
+
+class Put(HTTPRequestTypes):
+    @classmethod
+    def call(cls, url: TUrl, data: str) -> Response:
+        return put(url, data)
+
+
+class Delete(HTTPRequestTypes):
+    @classmethod
+    def call(cls, url: TUrl, _data: str) -> Response:
```
Comment (eriksendc) [inline hasher-matcher-actioner/hmalib/common/actioner_models.py:72]: ditto

**Review by eriksendc — APPROVED** [review]
Approving so not blocked, but please make the three changes I requested.

File: hasher-matcher-actioner/hmalib/common/actioner_models.py (hunk 2 above)
Comment (jeberl) [reply hasher-matcher-actioner/hmalib/common/actioner_models.py:43]: Couldn't find

**Review by eriksendc — APPROVED** [review]

**Pull request merged**